lib/ASTGen/Sources/ASTGen/SourceFile.swift

```diff
@@ -140,12 +140,20 @@ public func parseSourceFile(
     parsed = Syntax(MemberBlockItemListSyntax.parse(from: &parser))
 
   case .accessorMacroExpansion:
-    // FIXME: Implement specialized parsing.
-    parsed = Syntax(SourceFileSyntax.parse(from: &parser))
-  case .memberAttributeMacroExpansion,
-      .attribute:
-    // FIXME: Implement specialized parsing.
-    parsed = Syntax(SourceFileSyntax.parse(from: &parser))
+    parsed = Syntax(GeneratedAccessorBlockFileSyntax.parse(from: &parser))
+
+  case .memberAttributeMacroExpansion:
+    var attrs = GeneratedAttributeClauseFileSyntax.parse(from: &parser)
+    if !attrs.modifiers.isEmpty {
+      // 'memberAttribute' macro doesn't allow modifiers. Move to "unexpected" if any.
+      attrs.unexpectedBetweenAttributesAndModifiers = [Syntax(attrs.modifiers)]
+      attrs.modifiers = []
+    }
+    parsed = Syntax(attrs)
+
+  case .attributeFromClang:
+    parsed = Syntax(GeneratedAttributeClauseFileSyntax.parse(from: &parser))
+
   case .bodyMacroExpansion:
     // FIXME: Implement specialized parsing.
```

Comment:
Still thinking whether we need dedicated `Syntax` for this as well. It will be `CodeBlockItemList` + `endOfFile`, which is a subset of `SourceFileSyntax`. 
On the other hand, we probably need `generatedMemberBlockItemListFile` for `member` macro expansions.

Reply:
Correction, `body` macro emits a block (i.e. `{ ... }`) So we need a syntax node for it too. I went ahead and make `CodeBlockFileSyntax`